### PR TITLE
Adds transparency toggle

### PR DESCRIPTION
**Pull request by Cameron653**
- Adds a transparency toggle to prometheans. 
This allows them to become opaque for the people that are offput by the transparency of normal slimes.

Below: Various colors and species tested. 
T Human: https://i.imgur.com/n22nM0s.png
O Human: https://i.imgur.com/G4vquj3.png
Blue O Human: https://i.imgur.com/C7Awtiu.png
Blue T Human: https://i.imgur.com/CP2c8cl.png

T Teshari: https://i.imgur.com/F44S64A.png
O Teshari: https://i.imgur.com/lehB1KE.png

T Unathi: https://i.imgur.com/SI41ghz.png
O Unathi: https://i.imgur.com/d1gxWJW.png

T Nevrean: https://i.imgur.com/a60UL1V.png
O Nevrean: https://i.imgur.com/9ttWrJK.png

O Sergal: https://i.imgur.com/MXk8crF.png
T Sergal: https://i.imgur.com/uRtSg2n.png

### CODE REVIEW

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 1]
```diff
@@ -406,6 +406,8 @@ var/global/list/damage_icon_parts = list()
 				icon_key += "3"
 			else
 				icon_key += "1"
+			if(part.transparent) //VOREStation Edit. For better slime limbs.
+				icon_key += "_t" //VOREStation Edit. For better slime limbs.
```
Comment (Arokha) [inline code/modules/mob/living/carbon/human/update_icons.dm]: Comment #1

**Comment by Leshana** [comment]
Should work I imagine

**Comment by Cameron653** [comment]
Yeh. Fully tested, so I know for sure it works.

**Comment by Cameron653** [comment]
Any requests or anything? This is good to go, otherwise.

**Comment by Arokha** [comment]
I'm not a HUGE fan of this, given that it basically makes Prometheans into the "Well, just play them and you can make yourself look like literally any other species, why would you play anything else, given their mechanical advantages" sort of thing. No slip, immune to infections, no bleeding, no internal organs to damage, can regrow limbs, etc.

**Comment by Cameron653** [comment]
The reason you would want to play other species is the perks that you can't get by being a promethean, such as increased max health, increased speed, increased speed when using 'heavy' suits, burn resistance, flight, etc. 

While the prometheans have a lot of "traits", there are a long list of things that they can't get. Not to mention people can't play their normal cat/etc as a promethean and claim they're just a normal fox and then use the promethean traits, that'd be powergaming & immersion breaking.

Ontop of that, prometheans also have massive issues that other species don't. Regeneration takes a _quite_ considerable time. You can't just spam it like you used to be able to. That combined with their extremely weak limbs can easily make a wound incapacitate them much easier than any other species. 

They also have NO_SCAN. **This means they can't be resleeved.** This is a _massive_ downside that can't be recreated with any other species and makes up for all their upsides. One death and they're basically out of their original body and must be resleeved into a new body.

**Comment by Arokha** [comment]
> can't play their normal cat/etc as a promethean and claim they're just a normal fox and then use the promethean traits, that'd be powergaming & immersion breaking.

Given that you can just take a promethean, paint them a color, and since you made it so they can name themselves whatever, they can just put 'vulpkanin'... I mean who would know? You'd have to VV them to find out, and I don't really want to do that.

> They also have NO_SCAN

We've already told people they can respawn if they die, so I'm unsure how this hampers them. I'm also unsure why you think they can't be resleeved, because, they can be. NO_SCAN prevents cloning, which we don't use.

`toocomplex = (S.spawn_flags & SPECIES_IS_WHITELISTED) || (S.spawn_flags & SPECIES_IS_RESTRICTED)` is the cloning restriction. So whitelisted or restricted species can't be relseeved.

**Comment by Cameron653** [comment]
Well, you wouldn't know if someone was a promethean unless they were doing something like walking in space without any clothing and then claiming that they're fine and a normal vulp. Again, this isn't something people would really take just for the traits to powergame, especially seeing as it's an RP server and mechanics doesn't pop up too frequently outside of something like the expedition.

And while the 30 seconds is less than the xenochimera, it is still far more than when it used to be just spam-click it and be immune to damage. Hitting a leg can cause the limb to fall off and they're out for 30 seconds, while with humans/any other species you'd need to get extrordinarily lucky and take the limb off in 1 swipe or hit it multiple times to take it off. For reference, slime limbs have a _max_damage of 5_. I skimmed the code to see just what that means in context, and that means the organ should die and be completely unusable (due to being set to ORGAN_DEAD) if the limb takes more than 5 damage, provided that portion of the code isn't bugged.

https://github.com/VOREStation/VOREStation/blob/1df2ee75379811bbb914f1713cc02b6dd20db47d/code/modules/organs/organ.dm#L111-L113

https://github.com/VOREStation/VOREStation/blob/1df2ee75379811bbb914f1713cc02b6dd20db47d/code/modules/organs/organ.dm#L85-L89

https://github.com/VOREStation/VOREStation/blob/8ff288669d89f08ef326826df5d1fd4105dde3fe/code/modules/organs/organ_external.dm#L1133-L1134


I was under the impression that no_scan prevented resleeving, as I couldn't find any reference to it and when I asked I was told the body printer can't print species with no_scan. Ignore that point if that's not the case.

If people _really_ wanted to powergame by playing a promethean, they'd already be doing it. I'd imagine that a simple thing like transparency wouldn't affect them since they're playing to win (even though they do have downsides like extremely weaker limbs & higher burn damage). 

This just makes transparency toggleable for people that want to play a promethean made of opaque fluid/thick fluid, or have a custom species that uses the promethean as a base (such as natje's Alraune). This doesn't do anything like buff prometheans, it simply allows people that have a reason to turn the transparency off to do so.

**Comment by aztectornado** [comment]
I have no strong feelings about balance issues except that I feel like punishing on a case-by-case basis would work as opposed to assuming everyone is a powergamer.

That said, I like the idea of opaque shapeshifting things. This'd open up the idea of nanobot swarms, shapeshifting amalgamates, and so on, and I'm all for species with TF that's within reason. 

(Come to think of it, I kinda wish TF without vore bellies was a thing. Shapeshifting to other forms during scenes sounds hella fun compared to making a character for each and every single scenario you might want.)

**Comment by Verkister** [comment]
I'm gonna make some sorta oil/tar/sludge goop char when this gets in.

**Comment by Cameron653** [comment]
@Screemonster @JoanRisu @SpadesNeil @Leshana 

Can I get approval to merge/feedback on this? 

It'd be useful for non-transparent slime characters & for people like @Natje0 that have a SCA species that requires this for their character to look proper. 

I don't believe that people will start going promethean just for mechanical advantages, especially seeing as prometheans have a major amount of downsides, such as limbs that can instantly be removed in a single attack.

**Comment by Verkister** [comment]
I personally never saw the translucency as any sort of a "balance" feature for the prommies, only as a cosmetic slime aesthetic thing.

**Comment by DemgelZalvine** [comment]
To be honest, I'm not sure I like the idea of it. I'm also on the fence of 'why do we even have prommies at this point' because...tbqh, I think I'm the only one who plays an _actual promethean_. It feels almost like we should just cut the lore off them and rename them 'custom slime.' With this, there'll be even less reason to play a regular prommie. =w=; Not to mention with Proteans going to be a thing soon, I think the solid look should be left to them rather than adding it to prommies too. Want to look solid? apply for whitelist for nanogoop.

**Comment by Cameron653** [comment]
But not everyone wants to play a nanogoop, nor do they want to go through the effort to make a SCA just to have an opaque slime character, especially seeing as it'd most likely be extremely difficult to get on the list seeing all the abilities they have, some of which could be easily exploited.

Not only that, but proteans aren't made of slime, but nanites. Sure, it leaves for some uniqueness, but not nearly as much as slimes. 

There's not too many people play the base species. People name vulps/zorren foxes, unathi lizards, etc. 

Additionally, there is no actual lore for prometheans at the moment, along with nothing saying that they _have_ to be transparent.

**Comment by Screemonster** [comment]
tbh I'm fine with them having the option for cosmetic reasons, especially since the limb fix has made them no longer viable in any kind of "hee hee I'm going to pretend to not be a prommie and robust people who don't know I'm a prommie" 'cause whoever you fight is just gonna literally punch your face off.

**Comment by SpadesNeil** [comment]
I am inclined to agree with Aronai. I do not like this PR.

Gonna make the executive decision to close it. I see no point in even being any other species than Promethean if I can just recolor my skin to be human, except now I can also shapeshift.

**Pull request closed without merging**